src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -193,18 +193,20 @@ export class LocalKernelFinder implements ILocalKernelFinder {
             kernelSpecs.map(async (k) => {
                 // Find the interpreter that matches. If we find one, we want to use
                 // this to start the kernel.
-                const matchingInterpreters = this.findMatchingInterpreters(k, interpreters);
-                if (matchingInterpreters && matchingInterpreters.length) {
+                const matchingInterpreter = this.findMatchingInterpreter(k, interpreters);
+                if (matchingInterpreter) {
                     const result: PythonKernelConnectionMetadata = {
                         kind: 'startUsingPythonInterpreter',
                         kernelSpec: k,
-                        interpreter: matchingInterpreters[0],
-                        id: getKernelId(k, matchingInterpreters[0])
+                        interpreter: matchingInterpreter,
+                        id: getKernelId(k, matchingInterpreter)
                     };
 
                     // If interpreters were found, remove them from the interpreter list we'll eventually
                     // return as interpreter only items
-                    filteredInterpreters = filteredInterpreters.filter((i) => !matchingInterpreters.includes(i));
+                    filteredInterpreters = filteredInterpreters.filter(
+                        (i) => matchingInterpreter === i || this.fs.areLocalPathsSame(matchingInterpreter.path, i.path)
```

Comment:
This local path check doesn't look right to me. We have PythonEnvironments here, right? As far as I know (not the expert here) you can't match those by path. With conda without specifying a specific python multiple conda envs can share one local interpreter path.

Reply:
Fixed